### PR TITLE
Fix location info issues

### LINKED ISSUE #643: Crosshair is to true too often
If you connect to application: http://devv3.geoportail.lu/main/wsgi/theme/main
And then open the info panel by clicking on the info button in the bottom left bar. A "crosshair=true" parameter is added to the url. If you copy this url and paste it in a new tab, you'll see a marker in the map.

Moreover, if you right click somewhere in the map, the x and y coordinates in the url are for the clicked point (not the center of the map). Which is probably expected. But if you then close the info panel, x and y don't change back to the center of the map. If you share the link (using the share tool), the link doesn't show the same view.

### PR DESCRIPTION
**Pull request by petzlux**
fix #643

### CODE REVIEW

**Comment by petzlux** [comment]
@pgiraud  please review?

**Comment by jaykayone** [comment]
I rebased this, travis should now be OK....

File: geoportailv3/static/js/locationinfo/locationinfodirective.js [hunk 1]
```diff
@@ -111,10 +111,16 @@ app.LocationinfoController =
     }
     if (newVal === false) {
       this.stateManager_.updateState({'crosshair': false});
+      var mapCenterCoordinate = this['map'].getView().getCenter();
+      if (goog.isDefAndNotNull(mapCenterCoordinate)) {
```
Comment (pgiraud) [inline geoportailv3/static/js/locationinfo/locationinfodirective.js]: Is there any case where it can be null or undefined?

**Comment by pgiraud** [comment]
Sorry if I'm late reviewing this pull request but there's something I don't get with your code.
It looks like the main interest of the `coordinate_` property is to put a watcher on it and when it's not null set a new value to `location`. Why not modifying the `location` directly? (ie. in `showInfoPane_`)
I may be wrong but the code could be simplified a bit.

File: geoportailv3/static/js/locationinfo/locationinfodirective.js (hunk 1 above)
Comment (elemoine) [inline geoportailv3/static/js/locationinfo/locationinfodirective.js]: `getCenter`'s return type is `ol.Coordinate|undefined`. But given the way we use the View in the application `getCenter` should never return `null` or `undefined`. An assertion is therefore the best thing to use here.

```js
goog.asserts.assert(goog.isDefAndNotNull(viewCenter));
```

**Comment by petzlux** [comment]
@pgiraud , Regarding the coordinate_ variable, you were right, and I removed the variable.

Can you please check this branch on your machine, I have now issue after a rebase from master, that there is now a distortion of the map if the location info window gets closed. Before, as you know the inverse was the case, but you fixed that.

**Pull request merged**